lib/perl/Genome/Config/AnalysisProject/Command/AddEnvironmentFile.pm

```diff
@@ -0,0 +1,72 @@
+package Genome::Config::AnalysisProject::Command::AddEnvironmentFile;
+
+use strict;
+use warnings;
+
+use File::Spec;
+
+use Genome;
+
+class Genome::Config::AnalysisProject::Command::AddEnvironmentFile {
+    is => 'Genome::Config::AnalysisProject::Command::Base',
+    has_input => [
+        environment_file  => {
+            is                  => 'Path',
+            doc                 => 'path to the environment config file',
+            shell_args_position => 2,
+        },
+    ],
+};
+
+sub help_brief {
+    return 'add a yml environment file to an existing analysis project';
+}
+
+sub help_synopsis {
+    return "genome config analysis-project add-environment-file <analysis-project> <file-path>\n";
+}
+
+sub help_detail {
+    return <<"EOS"
+This command is used to link an environment configuration file to an analysis-project.  This specifies properties of the environment in which builds for the analysis-project will run.
+EOS
+}
+
+sub valid_statuses {
+    return ('Pending', 'In Progress', 'Hold');
+}
+
+sub execute {
+    my $self = shift;
+    my $analysis_project = $self->analysis_project;
+    if($analysis_project->environment_config_dir) {
+        $self->fatal_message('Project %s already has an existing environment file.', $analysis_project->__display_name__);
+    }
+
+    my $allocation = Genome::Disk::Allocation->get(owner => $analysis_project);
+    unless ($allocation) {
+        $allocation = Genome::Disk::Allocation->create(
```

Comment:
You're right, we should clean this up when it's deleted.  Maybe it does become an `ObjectWithAllocations`.

(Though in practice how often do we delete AnPs?)